base/abstractarray.jl

```diff
@@ -839,13 +860,13 @@ function hvcat{T}(rows::Tuple{Vararg{Int}}, as::AbstractMatrix{T}...)
                 throw(ArgumentError("mismatched height in block row $(i)"))
             end
             if c-1+szj > nc
-                throw(ArgumentError("block row $(i) has mismatched number of columns"))
+                throw(ArgumentError("block row $(i) has mismatched number of columns: $(c-1+szj) $nc"))
             end
```

Comment:
Would be clearer to me to write `block row $(i) has mismatched number of columns (expected $nc, got $(c-1+szj))`